task11/src/NaiveTree.hs

```diff
@@ -43,16 +44,27 @@ merge = undefined
   Скорее всего, при реализации вам потребуется функция 'merge'.
 -}
 instance Map NaiveTree where
-    empty = undefined
+    empty = Nil
 
-    singleton = undefined
+    singleton k a = Node k a Nil Nil
 
-    toAscList = undefined
+    toAscList Nil            = []
+    toAscList (Node k a l r) = (toAscList l) ++ ((k, a) : (toAscList r))
 
-    alter = undefined
+    alter f k Nil = maybe Nil (singleton k) (f Nothing)
+    alter f key (Node k a l r)
+                             | key == k = maybe (merge l r) (\val -> Node k val l r) (f (Just a))
```

Comment:
* Лучше сделать это последним случаем и назвать `otherwise`, чтобы warning'ов не было.
* Для создания вершины по значению есть готовая функция
* В скобочках для `f` лучше воспользоваться долларом